Dockerfile

```diff
@@ -29,11 +12,40 @@ RUN cargo +nightly install --git https://github.com/alexcrichton/wasm-gc
 # show backtraces
 ENV RUST_BACKTRACE 1
 
-# cleanup
-RUN apt autoremove -y
-RUN apt clean -y
-RUN rm -rf /tmp/* /var/tmp/*
+## not sure which of theses ENV are actually needed for this step
+#compiler ENV
+ENV CC gcc
+ENV CXX g++
+
+#snapcraft ENV
+ENV LC_ALL=C.UTF-8
+ENV LANG=C.UTF-8
+
+# Copy runtime library files
+COPY ./runtime/Cargo.lock ./runtime/Cargo.toml ./runtime/
+COPY ./runtime/src ./runtime/src
+# Copy WASM build crate files
+COPY ./runtime/wasm/build.sh ./runtime/wasm/Cargo.lock ./runtime/wasm/Cargo.toml ./runtime/wasm/
+COPY ./runtime/wasm/src ./runtime/wasm/src
 
+# get build script and build
+COPY ./scripts/build.sh /scripts/build.sh
+RUN /bin/bash /scripts/build.sh
+
+# this container builds the mashnet-node binary from source files, the runtime library and the WASM file built previously
+FROM rustlang/rust:nightly as builder
```

Comment:
couldn't you just use the same container and build the binaries?

Reply:
this would lead to earlier cache invalidation in case the runtime is changed, I think